sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/components/EmailGenerator.groovy

```diff
@@ -3,65 +3,99 @@ package life.qbic.samplenotificator.components
+     * Triggers the adaption of the emailTemplate to contain the information provided in the notificationContent
      */
-    private void prepareEmails(NotificationContent notificationContent) {
+    private Document fillEmailContent(NotificationContent notificationContent) {
+        emailHTMLTemplate = new EmailHTMLTemplate(Jsoup.parse(emailNotificationTemplateSupplier.get(), "UTF-8", ""))
         Document filledEmailTemplate = emailHTMLTemplate.fillTemplate(notificationContent)
-        emails.put(filledEmailTemplate, notificationContent.customerEmailAddress)
+        return filledEmailTemplate
+    }
+
+    /**
+     * Concatenates email header information and the actual email content into a singular file
+     *
+     * Method to create a file containing the email header information{@see notification-template/header.txt} and the dynamically adapted email message.
+     * This is necessary since sendmail doesn't provide a dedicated command line switch to set the content-type of the message
+     * @param notificationContent String representation of the filled in HTML Email Template {@see notification-template/email-update-template.html}
+     * @return concatenatedHTMLFile of the mailutils sendmail tool for detailed information see(@link <a href=https://www.cs.ait.ac.th/~on/O/oreilly/tcpip/sendmail/ch36_05.htm/>here</a>)
+     */
+    private File GroupEmailBodyAndHeaderIntoHTMLFile(String notificationContent) {
+        File concatenatedHTMLFile = File.createTempFile("HTMLEmail", ".html")
+        concatenatedHTMLFile.append(emailHeaderTemplateSupplier.get())
+        concatenatedHTMLFile.append(notificationContent)
+        concatenatedHTMLFile.deleteOnExit()
+        return concatenatedHTMLFile
     }
 
-    void sendEmails() {
-        emails.each { Document emailContent, String emailRecipient ->
-            def tempNotificationFile = new File('TempNotificationFile.html')
-            tempNotificationFile.write(emailContent.html())
-            ProcessBuilder builder = new ProcessBuilder("mail", "-s ${subject}", emailRecipient).redirectInput(tempNotificationFile)
+    /**
+     * Sends the prepared html file to the email address provided in the notificationContent via sendmail
+     *
+     * The email is send via a commandline call to the mailutils sendmail tool(@link <a href=https://mailutils.org/>mailutils</a>) via a ProcessBuilder
+     * The "-t" switch allows sendmail to extract information about the email sender, subject and content-type from the email header.
+     *
+     * @param emailHTMLFile Prepared HTMLFile containing a dedicated emailHeader and the filled in emailTemplate
+     * @param emailRecipient String representation of the subscribers email address to which the email should be send
+     * @return ExitCode of the mailutils sendmail tool for detailed information see(@link <a href=https://www.cs.ait.ac.th/~on/O/oreilly/tcpip/sendmail/ch36_05.htm/>here</a>)
+     *
+     */
+    //ToDo Move this into separate class
```

Comment:
Is this still a todo?

```suggestion
```

Reply:
I would prefer adding an issue. Otherwise we will only collect todos and they will be forgotten.

Reply:
Good point